pysquirrel/core.py

```diff
@@ -137,63 +135,40 @@ def _load(self) -> None:
                     }
                     self.data.append(cls(**region))
 
-    def _set_index(self) -> None:
-        """
-        Builds search index to be used to retrieve regions.
-        """
-        for region in self.data:
-            for field in fields(region):
-                key = self.search_index.setdefault(field.name, {})
-                value = getattr(region, field.name)
-                if field.name == "code":
-                    key[value] = region
-                else:
-                    if value in key:
-                        key[value].append(region)
-                    else:
-                        key[value] = [region]
-
     def _search(
         self,
         param: str,
         value: str | int,
     ) -> set[NUTSRegion | SRRegion]:
         """
-        Searches database index for one value of a parameter
+        Searches database for one value of a region field
         and returns a set of all matching result(s).
 
         :param param: field to be searched
         :param value: value(s) to be searched in the field
-
         """
-        results = set(
-            flatten(
-                [
-                    self.search_index[param][key]
-                    for key in self.search_index[param]
-                    if key == value
-                ]
-            )
-        )
-
+        results = set(flatten([i for i in self.data if getattr(i, param) == value]))
```

Comment:
Since you've simplified the searching logic I don't think you need the flatten here anymore. 
I think:
```suggestion
        return set(i for i in self.data if getattr(i, param) == value)
```

should be enough